jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/Request.java

```diff
@@ -295,62 +379,92 @@ default Transport getTransport()
      * @return this request object
      * @throws IOException if the file does not exist or cannot be read
      */
-    Request file(Path file, String contentType) throws IOException;
+    default Request file(Path file, String contentType) throws IOException
+    {
+        return this;
+    }
 
     /**
      * @return the user agent for this request
      */
-    String getAgent();
+    default String getAgent()
+    {
+        return null;
+    }
 
     /**
      * @param agent the user agent for this request (corresponds to the {@code User-Agent} header)
      * @return this request object
      */
-    Request agent(String agent);
+    default Request agent(String agent)
+    {
+        return this;
+    }
 
     /**
      * @param accepts the media types that are acceptable in the response, such as
      * "text/plain;q=0.5" or "text/html" (corresponds to the {@code Accept} header)
      * @return this request object
      */
-    Request accept(String... accepts);
+    default Request accept(String... accepts)
+    {
+        return this;
+    }
 
     /**
      * @return the idle timeout for this request, in milliseconds
      */
-    long getIdleTimeout();
+    default long getIdleTimeout()
+    {
+        return 0;
+    }
 
     /**
      * @param timeout the idle timeout for this request
      * @param unit the idle timeout unit
      * @return this request object
      */
-    Request idleTimeout(long timeout, TimeUnit unit);
+    default Request idleTimeout(long timeout, TimeUnit unit)
+    {
+        return this;
+    }
```

Comment:
These implementations are clearly wrong/incomplete?    Are they being added just for ease of testing?
Perhaps they should emit at least a debug or warning that they are unimplemented?

Reply:
Reverted to be abstract.